homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -168,79 +168,136 @@ def supported_features(self):
             return SUPPORT_VIERATV | SUPPORT_TURN_ON
         return SUPPORT_VIERATV
 
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "name": self.name,
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": DEVICE_MANUFACTURER,
+        }
+
+    def update(self):
+        """Retrieve the latest data and renew connection."""
+        if self._remote:
+            try:
+                self._muted = self._remote.get_mute()
+                self._volume = self._remote.get_volume() / 100
+                self._state = STATE_ON
+            except OSError:
+                self._state = STATE_OFF
+
+        old_remote = self._remote
+        try:
+            if self._app_id and self._encryption_key:
+                self._remote = RemoteControl(
```

Comment:
It would be good to extract the remote control handling to a separate class so we don't need to store host, port etc on the entity. The entity shouldn't need to care about the connection details.

Reply:
Should I put the remote inside this class or make it two separate things? I was thinking of keeping `remote` and `remote_data` as init parameters

Reply:
It would be nice if we didn't have to check in all the entity methods if remote is None before calling the remote method. I'd try to let the new class handle that and move the `send_key` method inside the new class too.